oommfc/drivers/hysteresisdriver.py

```diff
@@ -59,13 +59,31 @@ class HysteresisDriver(Driver):
     ]
 
     def _checkargs(self, **kwargs):
```

Comment:
This suggested change means your kwargs are already "dictionary-styled"?

Reply:
Accepted in https://github.com/ubermag/oommfc/pull/138/commits/b62fa9fc2acab598b9d218179a61517edc896ea0

Reply:
With the old concept using `**kwargs`, all is working fine with the other suggestions, out of the box.

Reply:
When you say everything works fine, does that imply both the old definition with `Hmin`, `Hmax` and `n` and the new definition with `Hsteps` or just the new `Hsteps`?

Reply:
I'll add any changes that might still be required for this after you have clarified what exactly you have tested already.

Reply:
I've tested both, providing once only `Hmin`, `Hmax` and `n` and the other time only `Hsteps` with the function header `def _checkargs(self, **kwargs):`. Using `kwargs` without the unpack operator `**` would lead to the necessity defining a dictionary as argument / input somewhere before `def_checkargs` is getting called.

Reply:
The relevant lines for testing it are:

```python
hd = oc.HysteresisDriver()
# driving with Hsteps
hd.drive(system, n_threads=4, Hsteps=[
    [(0, 0, 0), (5/mm.consts.mu0, 0, 0), 50],
    [(5/mm.consts.mu0, 0, 0), (2/mm.consts.mu0, 0, 0), 10],
    [(2/mm.consts.mu0, 0, 0), (0.03/mm.consts.mu0, 0, 0), 50],
])
# driving with Hmin, Hmax, n
hd.drive(system, n_threads=4, Hmin=(0.03/mm.consts.mu0, 0, 0), Hmax=(0.05/mm.consts.mu0, 0, 0), n=5)
```

Just two simple driving examples, both working as expected with `**kwargs`. If you need a full MWE please write me, I'm quite busy currently. Please be aware that I changed it in this branch of this pull request already to `kwargs`, so this version is not working for me.

Reply:
Ok I see ... maybe I missunderstood your first argument. I've not integrated (Hmin, Hmax, n) as a `list`object! Hmin, Hmax and n are (as in previous versions) valid keywords.

I've implemented `Hsteps` in such a way that your old concept providing three keywords, `Hmin=...`, `Hmax=...`, `n=...` is still fully valid and is not breaking with your concept in previous versions. Therefore, IMHO, `**kwargs` should remain.

Because of your great idea of implementing (Hmin, Hmax, n) as a symmetric version of `Hsteps` here in this file, your idea of a single code path in the scripts submodule is still working fine.

Reply:
Reverted `def _checkargs(self, kwargs):` to `def _checkargs(self, **kwargs):` in https://github.com/ubermag/oommfc/pull/138/commits/c089c7847c16bcd6df8d6f5a097ac40f1631c69e. This should be able to pass your current unittests.